src/lib.rs

```diff
@@ -465,4 +466,50 @@ mod tests {
         assert_eq!(first.received().unwrap(), 1463868507);
         teardown();
     }
+
+    #[test]
+    fn check_create_dirs() {
+        let maildir = Maildir::from("testdata/maildir2");
+        assert!(!Path::new("testdata/maildir2").exists());
+        assert!(!Path::new("testdata/maildir2/cur").exists());
+        assert!(!Path::new("testdata/maildir2/new").exists());
+        assert!(!Path::new("testdata/maildir2/tmp").exists());
+
+        maildir.create_dirs().unwrap();
+        assert!(Path::new("testdata/maildir2").exists());
+        assert!(Path::new("testdata/maildir2/cur").exists());
+        assert!(Path::new("testdata/maildir2/new").exists());
+        assert!(Path::new("testdata/maildir2/tmp").exists());
+
+        fs::remove_dir_all("testdata/maildir2").unwrap();
+    }
+
+    #[test]
+    fn check_store_new() {
+        let maildir = Maildir::from("testdata/maildir2");
+        maildir.create_dirs().unwrap();
+
+        assert_eq!(maildir.count_new(), 0);
+        maildir.store_new::<Box<std::error::Error>>("Return-Path: <of82ecuq@cip.cs.fau.de>
+X-Original-To: of82ecuq@cip.cs.fau.de
+Delivered-To: of82ecuq@cip.cs.fau.de
```

Comment:
Probably better to use fake email addresses here. Once it goes into the revision history it'll be there forever.